ansible-ipi-install/roles/installer/tasks/25a_disconnected_registry_check.yml

```diff
@@ -0,0 +1,43 @@
+---
+
+- name: Check if disconnected_registry_auths_file is set
+  set_fact:
+    dra_set: true
+  when:
+    - disconnected_registry_auths_file is defined
+    - disconnected_registry_auths_file is not none
+    - disconnected_registry_auths_file | trim != ''
+
+- name: Check if disconnected_registry_mirrors_file is set
+  set_fact:
+    drm_set: true
+  when:
+    - disconnected_registry_mirrors_file is defined
+    - disconnected_registry_mirrors_file is not none
+    - disconnected_registry_mirrors_file | trim != ''
+
+- name: Make sure disconnected_registry_variables are sane
+  fail:
+    msg:
+    - "Both variables must be set or unset."
+    - "  disconnected_registry_auths_file"
+    - "  disconnected_registry_mirrors_file"
+  when:
+        not ( dra_set and drm_set )
+    and not ( not dra_set and not drm_set )
+
+
+- name: Check if using existing disconnected registry.
+  set_fact:
+    disconnected_registry: existing
+  when:
+    - dra_set
+    - drm_set
+
+- name: Check if creating a new disconnected registry.
+  set_fact:
+    disconnected_registry: create
+  when:
+    - not dra_set
+    - not drm_set
+
```

Comment:
As the way the playbook stands today, this likely belongs in the 10_validation.yml. However, I probably need to figure out how to better modularize the existing playbook. Reason I point it out is, if theres a problem it would be great if user failed fast and 10_validation is one of the first tasks to run.

Reply:
placing this in the 10 validation playbook would make it run even when not deploying the registry. 
placing it here and only running it when creating a registry makes more sense to me.
These are checks that only need to be made when creating a registry.

Reply:
This is probably fine for now tbh. We could easily just add a when clause when registry host is set for all those checks in 10_validation and fail quicker but I think I'm going to change how that all works anyway so for now carry on

Reply:
i moved these validations to the node_prep role to be run as 15_* so any missing information causes a failure during the beginning of the playbook run

Reply:
in latest commit